### PR TITLE
bring ViewableRange and view::all back into alignment

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

**Comment by ericniebler** [comment]
Ok, I'm don't futzing with this.

File: include/stl2/detail/range/concepts.hpp [hunk 1]
```diff
@@ -265,7 +265,7 @@ STL2_OPEN_NAMESPACE {
 		template<class Rng>
 		concept bool ViewableRange =
 			Range<Rng> &&
-			(std::is_lvalue_reference_v<Rng> || View<__f<Rng>>);
+			(requires{typename __begin_t<Rng>;} || View<__f<Rng>>);
```
Comment (CaseyCarter) [inline include/stl2/detail/range/concepts.hpp]: `__begin_t` should be `__end_t` here, or require `Sentinel<__end_t<Rng>, __begin_t<Rng>>` directly.

File: include/stl2/view/all.hpp [hunk 2]
```diff
@@ -29,6 +30,13 @@ STL2_OPEN_NAMESPACE {
 				noexcept(std::is_nothrow_constructible_v<__f<Rng>, Rng>)
 				{ return std::forward<Rng>(rng); }
 
+				template <Range Rng>
+				requires !View<__uncvref<Rng>> &&
+					!std::is_reference_v<Rng> &&
+					requires {typename __begin_t<Rng>;}
```
Comment (CaseyCarter) [inline include/stl2/view/all.hpp]: Ditto.

**Comment by ericniebler** [comment]
OK now?

**Pull request merged**